opal/mca/common/ucx/common_ucx.c

```diff
@@ -39,9 +52,27 @@ OPAL_DECLSPEC void opal_common_ucx_mca_register(void)
                           MCA_BASE_VAR_TYPE_INT, NULL, 0, MCA_BASE_VAR_FLAG_SETTABLE,
                           OPAL_INFO_LVL_9, MCA_BASE_VAR_SCOPE_LOCAL,
                           &opal_common_ucx.progress_iterations);
+    mca_base_var_register("opal", "opal_common", "ucx", "opal_mem_hooks",
+                          "Use OPAL memory hooks, instead of UCX internal memory hooks",
+                          MCA_BASE_VAR_TYPE_BOOL, NULL, 0, 0,
+                          OPAL_INFO_LVL_3,
+                          MCA_BASE_VAR_SCOPE_LOCAL,
+                          &opal_common_ucx.opal_mem_hooks);
 
     opal_common_ucx.output = opal_output_open(NULL);
     opal_output_set_verbosity(opal_common_ucx.output, opal_common_ucx.verbose);
+
+    /* Set memory hooks */
+    if (opal_common_ucx.opal_mem_hooks &&
+        (OPAL_MEMORY_FREE_SUPPORT | OPAL_MEMORY_MUNMAP_SUPPORT) ==
+        ((OPAL_MEMORY_FREE_SUPPORT | OPAL_MEMORY_MUNMAP_SUPPORT) &
+         opal_mem_hooks_support_level()))
+    {
+        MCA_COMMON_UCX_VERBOSE(1, "%s", "using OPAL memory hooks as external events");
+        mca_base_framework_open(&opal_memory_base_framework, 0);
```

Comment:
need to call this before using opal_mem_hooks_support_level()

Reply:
fixed